src/collision.cpp

```diff
@@ -550,6 +550,10 @@ collisionMoveResult collisionMoveSimple(Environment *env, IGameDef *gamedef,
 					speed_f->X = 0;
 				result.collides = true;
 			} else if (nearest_collided == COLLISION_AXIS_Y) {
+				// small hack. we just bounced up, so we can jump.
+				if(speed_f->Y < 0.0 && bounce <  -0.001f)
+					result.touching_ground = true;
```

Comment:
Might it be possible to offload this to a new parameter? `touching_ground` must only indicate whether the object is actually colliding with ground to avoid follow-up issues.

Reply:
it seemed the best way to communicate the state with `localplayer.cpp`

normally the next cycle would detect the object within whatever tiny tolerance indicates passive collision and set touching_ground then. this code makes the detection instantaneous instead, but only for bouncy floors.

the only potential issue I found while examining the code is if we land on **another entity** which has a bouncy value. is that even possible?

a new variable would require communicating it with localplayer and ensuring it gets reset at the appropriate stage of each cycle. is this really necessary?

Reply:
The idea behind this suggestion was to clearly separate `touching_ground` from "can_my_player_jump" to avoid possible side-effects from this change. It generally leaves a bad impression when there's "hack"y code, especially when it affects existing physics calculations.

Also: Isn't `we just bounced up` inaccurate here? The bounce effect is yet to happen in the next few lines.

Reply:
Pre-existing code already seems to use `touching_ground` to imply some things. Example being it is set when standing on another CAO, and is already the main condition for jumping eligibility (`client/localplayer.cpp L448`).

My thinking here is that, since we struck a bouncy floor, we ARE touching the ground, but current physics code won't recognize it until the next tick. (I feel tempted to argue that it should be set instantly no matter what type of floor we hit, but I'm not fully familiar with the possible side effects of that, considering the variety of things that could be considered a floor)

I think to pedantically separate only "my feet are planted on a solid node" would suggest a little bit (not a _whole_ lot) of refactoring of code unrelated to this PR, and likely introduce more than one extra data member.

As for the comment, I can change it to "we are currently bouncing up" to make it more precise.

Reply:
FYI this is what prevented me from approving this PR before, messing with the data reported by collision to get some minor detail working is not okay (the wrong way)